### ISSUE TITLE
Grid: content found multiple times when it is actually a merged row

### ISSUE THREAD

**Issue by githubIsBugtrack** [mvysny/karibu-testing#52]
Using a setup like in this one: https://github.com/mvysny/t-shirt-shop-example :
Spring boot, Java, Maven.
Version: 1.2.5

I use a grid which has a row above the actual header, like this (sorry for huge whitespace) :
![Example](https://user-images.githubusercontent.com/61277292/95861922-6b1dc000-0d62-11eb-96fe-fd8653d28d38.jpg)
Unfortunately, "myHeading" and "myButton" get found 3 times  (i assume once per column), while this first line of the grid is actually one row with 1 heading and 1 button. Browser confirms. This occurs while prettyPrinting as well as for the actual com.github.mvysny.kaributesting.v10.LocatorJ._get(Button.class, spec -> spec.withCaption("myButton")) method.
I took the workaround with LocatorJ._find(..) which returns a List and now just take the first entry/button to klick on it, but if you could tell me what I have to do in order to get rid of the workaround, or if it is a bug, please let me know. 

Best regards

Rene

**Comment by mvysny** [comment]
Hi Rene, could you please paste the output of prettyPrint() here? That would tell me where exactly the component is duplicated three times, and I will be able to reproduce the issue easier. Please wrap the output within the triple back-tick block so that it's readable:

```
pretty print output
```

**Comment by githubIsBugtrack** [comment]
Hei Martin,

thanks for your response. 
I can give you a prettyprint with some things "censored", if that is ok?
The Grid at that time is empty, therefore no lines besides the header.
Please bear in mind: The lack of "heading-header" is what I tried to describe as a possible bug, so can't be found in that print.
```

Component tree:
└── UI[]
... some other stuff goes here...
    │   ├── Div[width:100% height:100% overflow:auto]
    │   │   └── MyView[width:100% height:100%]
    │   │       └── MyCustomFieldClass[width:100% height:100%, value='[]']
    │   │           └── FlexLayout[width:100% align-items:stretch display:flex height:100%]
		..more stuff ..
    │   │                   └── VerticalLayout[slot='secondary', width:100% height:100%]
    │   │                       └── Grid[width:100% height:100%, dataprovider='com.vaadin.flow.data.provider.DataProvider$2@1fc4b8d6']
    │   │                           └── ColumnGroup[]
    │   │                               ├── Column[header='Colour']
    │   │                               │   └── Leiste[width:100% padding:0 justify-content:space-between min-height:50px align-items:center]
    │   │                               │       ├── Span[margin-left:0 font-size:var(--lumo-font-size-xl) position:absolute, text='MyHeader']
    │   │                               │       └── Button[cursor:pointer, caption='MyButton', icon='vaadin:plus']
    │   │                               │           └── Icon[slot='prefix', icon='vaadin:plus']
    │   │                               ├── Column[header='Nachname']
    │   │                               │   └── Leiste[width:100% padding:0 justify-content:space-between min-height:50px align-items:center]
    │   │                               │       ├── Span[margin-left:0 font-size:var(--lumo-font-size-xl) position:absolute, text='MyHeader']
    │   │                               │       └── Button[cursor:pointer, caption='MyButton', icon='vaadin:plus']
    │   │                               │           └── Icon[slot='prefix', icon='vaadin:plus']
    │   │                               └── Column[header='Vorname']
    │   │                                   └── Leiste[width:100% padding:0 justify-content:space-between min-height:50px align-items:center]
    │   │                                       ├── Span[margin-left:0 font-size:var(--lumo-font-size-xl) position:absolute, text='MyHeader']
    │   │                                       └── Button[cursor:pointer, caption='MyButton', icon='vaadin:plus']
    │   │                                           └── Icon[slot='prefix', icon='vaadin:plus']
    │   │                              
...some other stuff goes here ...
```

**Comment by mvysny** [comment]
Perfect, thank you. Could you also post the code which creates columns for the Grid please? I can see that there is a ColumnGroup which indicates that there could be multiple header bars?

**Comment by githubIsBugtrack** [comment]
Ok, so here is some minimalistic code which reproduces the occurrence of the button.


Testcode:
```
    @Test
    public void testMinimalXample() {
        UI.getCurrent().navigate(MinimalXample.class);
        _get(Button.class, capture -> capture.withCaption("tykky"));
    }
```


Grid:

```
@Route("test")
public class MinimalXample extends VerticalLayout {

    public MinimalXample() {
        Grid<Bean> grid = new Grid<>();
        grid.setItems(new ArrayList<>());
        Column<Bean> columnColour = grid.addColumn(Bean::getColour).setHeader("Colour");
        Column<Bean> columnNumber = grid.addColumn(Bean::getNumber).setHeader("Number");
        Column<Bean> columnWeather = grid.addColumn(Bean::getWeather).setHeader("Weather");
        HeaderRow header = grid.prependHeaderRow();
        HeaderCell joinedCell = header.join(columnColour, columnNumber, columnWeather);
        joinedCell.setComponent(new Button("MyButton"));
        add(grid);
    }

    private class Bean {
        private String colour;
        private String number;
        private String weather;
        public String getColour() {
            return colour;
        }
        public void setColour(String colour) {
            this.colour = colour;
        }
        public String getNumber() {
            return number;
        }
        public void setNumber(String number) {
            this.number = number;
        }
        public String getWeather() {
            return weather;
        }
        public void setWeather(String weather) {
            this.weather = weather;
        }
    }
}

```



```
java.lang.AssertionError: /test: No visible Button in UI[] matching Button and caption='tykky': []. Component tree:
└── UI[]
    └── MinimalXample[width:100%]
        └── Grid[dataprovider='com.vaadin.flow.data.provider.ListDataProvider@4bec6682']
            └── ColumnGroup[]
                ├── Column[header='Colour']
                │   └── Button[caption='MyButton']
                ├── Column[header='Number']
                │   └── Button[caption='MyButton']
                └── Column[header='Weather']
                    └── Button[caption='MyButton']
```

**Comment by mvysny** [comment]
Awesome, thank you :+1:  Yes, it was exactly as I suspected: a component from a joined header cell is reported multiple times. I have changed the code so that the component is only present once, and therefore you will be able to use `_get`. The fix will appear in Karibu-Testing 1.2.6.

**Comment by githubIsBugtrack** [comment]
Wow, you're fast! :D Thanks a lot, Martin!
